src/date-picker/date-picker.jsx

```diff
@@ -77,6 +77,11 @@ let DatePicker = React.createClass({
       ...other,
     } = this.props;
 
+    // Removes valueLink to avoid `Invariant Violation: Cannot provide a valueLink
+    // and a value or onChange event. If you want to use value or onChange, you
+    // probably don't want to use valueLink.`
+    if (other.valueLink) delete other.valueLink;
```

Comment:
A better way to do this would be to add `valueLink` inside the spread operator. However, in looking at the code, I don't think this component is handling `valueLink` properly since it should return the date as the first parameter in the `onChange` callback. Also, it should call valueLink.requestChange. This will have to be addressed in a separate PR.